homooligomer_rmsd.py

```diff
@@ -0,0 +1,153 @@
+    structure_target, structure_mobile = align_chain_A(path_to_target, path_to_mobile, parser)
+    target_chains = list(structure_target.get_chains())
+    mobile_chains = list(structure_mobile.get_chains())
+
+    target_chains_com = chains_com_coords(target_chains)
+    mobile_chains_com = chains_com_coords(mobile_chains)
+
+    distances_array = []
+    for key_rfdiff in target_chains_com:
+        distances_array.append([])
+        for key_af in mobile_chains_com:
+            dst = distance.euclidean(target_chains_com[key_rfdiff], mobile_chains_com[key_af])
+            distances_array[-1].append(dst)
+
+    #get dictionaries of corresponding chain names 
+    corresponding = np.argmin(distances_array, axis=1)
+    rename_dict_1 = {}
+    rename_dict_2 = {}
+    for i, j in zip(corresponding, range(len(corresponding))):
+        str1=list(target_chains_com.keys())
+        str2=list(mobile_chains_com.keys())
+        str3=[n.lower() for n in str2]
+        rename_dict_1[str1[i]]=str3[j]
+        rename_dict_2[str3[j]]=str2[j]
+
+    #rename chains of mobile structure
+    #if the structures are very different, multiple chains in mobile can be close to target 
+    #in that case, return rmsd = -1
+    try:
+        for model in structure_mobile:
+            for chain in model:
+                old_name = chain.get_id()
+                chain.id = rename_dict_1[old_name]
+        for model in structure_mobile:
+            for chain in model:
+                old_name = chain.get_id()
+                chain.id = rename_dict_2[old_name] 
+    except KeyError:
+        rmsd = -1
+        if save_aligned==True:
+            save_structures(structure_target, structure_mobile, rmsd) 
+        return rmsd
+    
+    #get a list of residues for alignment for mobile structure
+    target_res_total = list(structure_target.get_residues())
+    mobile_res_total = []
+    for i in list(structure_target.get_chains()):
+        for k in mobile_chains:
+            if k.id == i.id:
+                mobile_res_total += list(k.get_residues())   
+    target_res_total = [ind['CA'] for ind in target_res_total]
+    mobile_res_total = [ind['CA'] for ind in mobile_res_total]
+
+    superimposer = Superimposer()
+    superimposer.set_atoms(target_res_total, mobile_res_total)
+    superimposer.apply(structure_mobile.get_atoms())
+    rmsd = superimposer.rms
+
+    if save_aligned==True:
+        save_structures(structure_target, structure_mobile, rmsd)
+
+    return rmsd
+
+
+if __name__ == '__main__':
+    # Define the command-line arguments
+    argparser = argparse.ArgumentParser(description='calculate rmsd of two oligomers')
+    argparser.add_argument('--path_to_target', '-t', type=str,
+                        help='path to the pdb of target protein (the one to align to)')
+    argparser.add_argument('--path_to_mobile', '-m',  type=str,
+                        help='path to the pdb of mobile protein')
+    # Parse the arguments
```

Comment:
Save aligned could be a flag that is passed.
Also rmsd should be printed to stdout.